src/ValidationManifold.jl

```diff
@@ -127,31 +237,35 @@ function copyto!(
     return p
 end
 
+decorated_manifold(M::ValidationManifold) = M.manifold
+
 function distance(M::ValidationManifold, p, q; kwargs...)
-    is_point(M, p; error = M.mode, kwargs...)
-    is_point(M, q; error = M.mode, kwargs...)
-    return distance(M.manifold, array_value(p), array_value(q))
+    _vMc(distance, (:Point, :Input), M.ignore) && is_point(M, p; error = M.mode, kwargs...)
```

Comment:
I'd suggest putting `is_point` check in `_vMc` or some other helper function so that less code is duplicated across functions.

Reply:
The call in `distance` would be reduced to just
```julia
 _vMc_is_point(M, distance, :Input, p; kwargs...)
```
instead of
```julia
_vMc(distance, (:Point, :Input), M.ignore) && is_point(M, p; error = M.mode, kwargs...)
```
which looks much shorter.

Reply:
yes, but I would implement quite a few such wrappers?

Reply:
Only for points and vectors, so two? Other things wouldn't necessarily need such wrapper because they are more rare.

Reply:
hm, will think about it, it is a bit less flexible then

Reply:
`_vMc` could still remain as is but it would have a less flexible wrapper for easier usage in typical cases.

Reply:
fair enough